### PR TITLE
Avoid wrapping nil input

### PR DESCRIPTION
**Pull request by tt**
Rack ships with the `Rack::RewindableInput::Middleware` middleware for compatibility with older apps.

However, this middleware wraps `rack.input` unconditionally meaning that a nil input will also be wrapped potentially leading to exceptions.

Concretely, I have an older app with a middleware that's trying to `req.body&.read` and `req.body&.rewind` which for `GET` and `HEAD` requests now cause the following error:

```
NoMethodError: undefined method `read' for nil
   ~/.gem/ruby/3.3.6/gems/rack-3.1.8/lib/rack/rewindable_input.rb:96:in `make_rewindable'
   ~/.gem/ruby/3.3.6/gems/rack-3.1.8/lib/rack/rewindable_input.rb:41:in `read'
```

This middleware should probably interrogate `req.request_method`.

Even so, it seems that for backwards compatibility (and to avoid the possibility of runtime errors overall), it would be great to never wrap a nil value.

### CODE REVIEW

File: test/spec_rewindable_input.rb [hunk 1]
```diff
@@ -167,9 +167,16 @@ def @io.rewind
 
 describe Rack::RewindableInput::Middleware do
   it "wraps rack.input in RewindableInput" do
-    app = proc{|env| [200, {}, [env['rack.input'].class.to_s]]}
-    app.call('rack.input'=>StringIO.new(''))[2].must_equal ['StringIO']
+    app = proc { |env| [200, {}, [env['rack.input'].class.to_s]] }
+    app.call('rack.input' => StringIO.new(''))[2].must_equal ['StringIO']
     app = Rack::RewindableInput::Middleware.new(app)
-    app.call('rack.input'=>StringIO.new(''))[2].must_equal ['Rack::RewindableInput']
+    app.call('rack.input' => StringIO.new(''))[2].must_equal ['Rack::RewindableInput']
```
Comment (tt) [inline test/spec_rewindable_input.rb:173]: Happy to undo these changes if desired and potentially also drop the spaces in the new test. It looked like newer code preferred this syntax.

File: lib/rack/rewindable_input.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@ def initialize(app)
       end
 
       def call(env)
-        env[RACK_INPUT] = RewindableInput.new(env[RACK_INPUT])
+        env[RACK_INPUT] = RewindableInput.new(env[RACK_INPUT]) unless env[RACK_INPUT].nil?
```
Comment (ioquatix) [inline lib/rack/rewindable_input.rb:24]:
```suggestion
        if input = env[RACK_INPUT]
          env[RACK_INPUT] = RewindableInput.new(input)
        end
```

How about avoiding extra hash lookups?

File: test/spec_rewindable_input.rb (hunk 1 above)
Comment (ioquatix) [reply test/spec_rewindable_input.rb:173]: For the sake of back porting and review, please don't include unrelated changes.

Comment (tt) [reply test/spec_rewindable_input.rb:173]: @ioquatix, want me to drop spaces in the new test for consistency inside this file then?

Comment (tt) [reply test/spec_rewindable_input.rb]: I removed the spaces. It does add a couple new RuboCop offenses so let me know if you prefer it to be different.

**Review by ioquatix — APPROVED** [review]
This looks okay to me.

**Review by jeremyevans — APPROVED** [review]

**Comment by ioquatix** [comment]
Thanks for your contribution. I hope we can release this in Rack 3.2 sometime before the end of this year.
